python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,374 @@
+            import pytorch_lightning as pl
+            from ray.air.config import ScalingConfig
+            from ray.train.lightning import LightningTrainer, LightningConfig
+
+            class MNISTClassifier(pl.LightningModule):
+                def __init__(self, lr, feature_dim):
+                    super(MNISTClassifier, self).__init__()
+                    self.fc1 = torch.nn.Linear(28 * 28, feature_dim)
+                    self.fc2 = torch.nn.Linear(feature_dim, 10)
+                    self.lr = lr
+                    self.accuracy = Accuracy()
+
+                def forward(self, x):
+                    x = x.view(-1, 28 * 28)
+                    x = torch.relu(self.fc1(x))
+                    x = self.fc2(x)
+                    return x
+
+                def training_step(self, batch, batch_idx):
+                    x, y = batch
+                    y_hat = self(x)
+                    loss = torch.nn.functional.cross_entropy(y_hat, y)
+                    self.log('train_loss', loss)
+                    return loss
+
+                def validation_step(self, val_batch, batch_idx):
+                    x, y = val_batch
+                    logits = self.forward(x)
+                    loss = F.nll_loss(logits, y)
+                    acc = self.accuracy(logits, y)
+                    return {"val_loss": loss, "val_accuracy": acc}
+
+                def validation_epoch_end(self, outputs):
+                    avg_loss = torch.stack([x["val_loss"] for x in outputs]).mean()
+                    avg_acc = torch.stack([x["val_accuracy"] for x in outputs]).mean()
+                    self.log("ptl/val_loss", avg_loss)
+                    self.log("ptl/val_accuracy", avg_acc)
+
+                def configure_optimizers(self):
+                    optimizer = torch.optim.Adam(self.parameters(), lr=self.lr)
+                    return optimizer
+
+            # Prepare MNIST Datasets
+            transform = transforms.Compose([
+                transforms.ToTensor(),
+                transforms.Normalize((0.1307,), (0.3081,))]
+            )
+            mnist_train = MNIST(
+                './data', train=True, download=True, transform=transform
+            )
+            mnist_val = MNIST(
+                './data', train=False, download=True, transform=transform
+            )
+            train_loader = DataLoader(mnist_train, batch_size=32, shuffle=True)
+            val_loader = DataLoader(mnist_val, batch_size=32, shuffle=False)
+
+            config_builder = LightningConfigBuilder()
+            config_builder.set_module_class(MNISTClassifier)
+            config_builder.set_module_init_config(lr=1e-3, feature_dim=128)
+            config_builder.set_trainer_init_config(max_epochs=5, accelerator="gpu")
+            config_builder.set_trainer_fit_params(
+                train_dataloaders=train_loader, val_dataloaders=val_loader
+            )
+            lightning_config = config_builder.build()
```

Comment:
Can we show the chained builder? Also, does this grouping of params make more sense?

```python
lightning_config = LightningConfigBuilder().module(class=MNISTClassifier, init_config={}).trainer(init_config={}, fit_params={}).build()
```

Reply:
Agree!